src/services/oauth2.js

```diff
@@ -36,22 +38,24 @@ export class OAuth2Service {
   updateUser(user, data) {
     const idField = this.options.user.idField;
     const id = user[idField];
+    const userService = typeof this._userService === 'string' ? this.app.service(this._userService) : this._userService;
```

Comment:
That's how it was originally setup, but if you do something like this:

```
app.configure(authentication);
app.use('/api/users', service(options));
```
The users service hasn't been setup by the time the auth plugin's `setup` method runs, it's still `undefined` until after the users service gets setup.  I made it more DRY by putting the lookup into a class method in that last commit.